### PR TITLE
stats: fix trigger condition for auto analyze

### PR DESCRIPTION
**Pull request by lamxTyler**
Because of the load by need strategy, if we have a table with no index, we may auto analyze it multiple times.

### CODE REVIEW

File: statistics/update_test.go [hunk 1]
```diff
@@ -289,6 +291,21 @@ func (s *testStatsUpdateSuite) TestAutoUpdate(c *C) {
...
+	h.DumpStatsDeltaToKV()
+	h.Clear()
+	// We set `Lease` here so that `Update` will use load by need strategy.
+	h.Lease = time.Second
+	h.Update(is)
+	h.Lease = 0
+	err = h.HandleAutoAnalyze(is)
+	c.Assert(err, IsNil)
+	h.Update(is)
+	stats = h.GetTableStats(tableInfo.ID)
```
Comment (shenli) [inline statistics/update_test.go]: Which assert makes sure that Analyze is not run multiple times?

Comment (lamxTyler) [reply statistics/update_test.go]: In line 307, we assert that the modify count is 1.

Comment (shenli) [reply statistics/update_test.go]: Please comment it.

File: statistics/update_test.go [hunk 2]
```diff
@@ -298,7 +315,7 @@ func (s *testStatsUpdateSuite) TestAutoUpdate(c *C) {
 	h.HandleAutoAnalyze(is)
 	h.Update(is)
 	stats = h.GetTableStats(tableInfo.ID)
-	c.Assert(stats.Count, Equals, int64(1))
+	c.Assert(stats.Count, Equals, int64(2))
```
Comment (shenli) [inline statistics/update_test.go]: Why this is changed?

Comment (coocood) [reply statistics/update_test.go]: Another row is inserted.

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
Should we cherry-pick this to release-1.0?

**Comment by lamxTyler** [comment]
@shenli Do not need to cherry-pick, because release-1.0 do not have load by need.

**Pull request merged**